tests/test_mock_settings.cpp

```diff
@@ -47,4 +47,20 @@ TEST(Settings, can_have_get_mocked)
     ASSERT_EQ(MP_SETTINGS.get(test), proof);
 }
 
+TEST(Settings, set_aux_invalid_throws)
+{
+    EXPECT_THROW(MP_SETTINGS.set("client.secondary-name", ""),
+                 multipass::InvalidSettingsException);
+}
+
+TEST(Settings, set_aux_client_primary_name_null)
+{
+    EXPECT_NO_THROW(MP_SETTINGS.set("client.primary-name", ""));
+}
+
+TEST(Settings, set_aux_client_primary_name_not_null)
+{
+    EXPECT_NO_THROW(MP_SETTINGS.set("client.primary-name", "not null name"));
+}
+
```

Comment:
The situation with `Settings` and tests is particular... We don't really have tests for the real `Settings` singleton, because it relies on `QSettings` which is not mockable. We'd have to provide another wrapper for `QSettings` which we could extend and mock and we never got around to that. 

OTH, we don't want tests to affect or be affected by real settings on the disk, so we make sure [all the tests run with a `MockSettings`](https://github.com/canonical/multipass/blob/c3d3afa0d62b9e3a766a769bd72d3d5f4ea84ac2/tests/main.cpp#L36) instead. This file has only a couple of very shallow tests for the mock itself, to show that it works when accessing the `Settings` singleton.

In short, what you wrote here only calls a mock and we don't have a way to actually test this ATM, so you can remove these entirely.

Reply:
Got it, thanks for the explanation!